sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java

```diff
@@ -75,23 +73,31 @@
     static final String ACCEPT_LANGUAGE = "en-US";
     static final int DEFAULT_MAX_PAGE_RESULTS = 25;
     static final String CONTENT_TYPE_HEADER_VALUE = "application/json";
-    private final String endpoint;
+    private final String vaultUrl;
     private final CertificateService service;
     private final ClientLogger logger = new ClientLogger(CertificateAsyncClient.class);
 
     /**
      * Creates a CertificateAsyncClient that uses {@code pipeline} to service requests
      *
-     * @param endpoint URL for the Azure KeyVault service.
+     * @param vaultUrl URL for the Azure KeyVault service.
      * @param pipeline HttpPipeline that the HTTP requests and responses flow through.
      * @param version {@link CertificateServiceVersion} of the service to be used when making requests.
      */
-    CertificateAsyncClient(URL endpoint, HttpPipeline pipeline, CertificateServiceVersion version) {
-        Objects.requireNonNull(endpoint, KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.VAULT_END_POINT_REQUIRED));
-        this.endpoint = endpoint.toString();
+    CertificateAsyncClient(URL vaultUrl, HttpPipeline pipeline, CertificateServiceVersion version) {
+        Objects.requireNonNull(vaultUrl, KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.VAULT_END_POINT_REQUIRED));
+        this.vaultUrl = vaultUrl.toString();
         this.service = RestProxy.create(CertificateService.class, pipeline);
     }
 
+    /**
+     * Get the vault endpoint url to which service requests are sent to.
+     * @return the vault endpoint url
+     */
+    public String getVaultUrl() {
+        return vaultUrl;
+    }
```

Comment:
Is there a need for this?

Reply:
Once the client is built and locked to an endpoint
There is no way to find out which vault url it points to.
If user has multiple clients set up, this can help to figure out different vault clients.

Plus, this change was standardized across all languages.